### ISSUE TITLE
Refactor lib.planner.onion

### ISSUE THREAD

**Issue by charredUtensil** [charredUtensil/hognose#2]
The current implementation generates roughly uniform layers throughout, and often produces unacceptably large rings of Hard Rock around larger caverns. Refactor this to something that produces better results. This will include:

- Remove "gradient" verbiage throughout
- Have layers that "sprinkle" i.e. Dirt and Loose Rock together.
- Some consideration for the size of the cave. Some layers make more sense in terms of tile numbers while others make sense as percentages

**Comment by charredUtensil** [comment]
What I've figured out so far is that the terminology I should be using here isn't "onion" - it's a [baroque pearl](https://en.wikipedia.org/wiki/Baroque_pearl). The trick I've added to the experimental branch starts with a well defined base layer, then walks around it in successive layers. These layers have tiny variances in direction which makes each layer progressively lumpier than the rest. The result is promising:

![image](https://github.com/charredUtensil/hognose/assets/1398543/dca1d1cc-973d-4858-a34e-3c9b035ecad0)

Right now, it's biasing to the top-right because of how it's implemented, and layers don't necessarily complete.
